site/src/modules/templates/useWatchVersionLogs.ts

```diff
@@ -0,0 +1,42 @@
+import { watchBuildLogsByTemplateVersionId } from "api/api";
+import { ProvisionerJobLog, TemplateVersion } from "api/typesGenerated";
+import { useState, useEffect } from "react";
+
+export const useWatchVersionLogs = (
+  templateVersion: TemplateVersion | undefined,
+  options?: { onDone: () => Promise<unknown> },
+) => {
+  const [logs, setLogs] = useState<ProvisionerJobLog[]>();
+  const templateVersionId = templateVersion?.id;
+  const templateVersionStatus = templateVersion?.job.status;
+
+  useEffect(() => {
+    setLogs(undefined);
```

Comment:
`undefined` is not a valid `ProvisionerJobLog[]`

Reply:
When you do this `useState<ProvisionerJobLog[]>();`, it infers you can also add `undefined` but I can make it more clear.